File: src/main/java/no/nav/domain/pensjon/regler/Opptjening.java
package no.nav.domain.pensjon.regler;

import no.nav.domain.pensjon.regler.beregning.Poengtall;
import no.nav.domain.pensjon.regler.beregning2011.LonnsvekstInformasjon;

import java.io.Serializable;

/**
 * Dataholderklasse for opptjening. Objektet vil være både input og output fra regeltjenester.
 */
public class Opptjening implements Serializable {
    private int ar;
    private double opptjeningsgrunnlag;
    private double anvendtOpptjeningsgrunnlag;
    private double arligOpptjening;
    private LonnsvekstInformasjon lonnsvekstInformasjon;
    private double pSatsOpptjening;
    private Poengtall poengtall;
    private double inntektUtenDagpenger;
    private Uforeopptjening uforeOpptjening;
    private double dagpenger;
    private double dagpengerFiskerOgFangstmenn;
    private double omsorg;
    private double forstegangstjeneste;
    private double arligOpptjeningOmsorg;
    private double arligOpptjeningUtenOmsorg;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/Opptjening.kt
package no.nav.pensjon.regler.domain

import no.nav.pensjon.regler.domain.beregning.Poengtall
import no.nav.pensjon.regler.domain.beregning2011.LonnsvekstInformasjon
import java.io.Serializable

/**
 * Dataholderklasse for opptjening. Objektet vil være både input og output fra regeltjenester.
 */
class Opptjening : Serializable {
    var ar = 0
    var opptjeningsgrunnlag = 0.0
    var anvendtOpptjeningsgrunnlag = 0.0
    var arligOpptjening = 0.0
    var lonnsvekstInformasjon: LonnsvekstInformasjon? = null
    var pSatsOpptjening = 0.0
    var poengtall: Poengtall? = null
    var inntektUtenDagpenger = 0.0
    var uforeOpptjening: Uforeopptjening? = null
    var dagpenger = 0.0
    var dagpengerFiskerOgFangstmenn = 0.0
    var omsorg = 0.0
    var forstegangstjeneste = 0.0
    var arligOpptjeningOmsorg = 0.0
    var arligOpptjeningUtenOmsorg = 0.0
}